### PR TITLE
WIP: Extract keyboard

### PR DESCRIPTION
**Pull request by nickdown**
This PR refactors Synthesizer.vue to accept frequencies from a new MusicalTyping component.
The later component displays a keyboard on the screen.

Still a WIP. could use some help on properly laying out the on screen keyboard

image

### CODE REVIEW

File: src/components/MusicalTyping.vue [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    components: {KeyButton},
+    props: ['octave'],
+    data() {
+      return {
+        activeFrequencies: {},
+        keys: [
+          {
+            letter: 'a',
+            pianoKey: 'C',
+            offset: 4,
```
Comment (FrankyFrankFrank) [inline src/components/MusicalTyping.vue]: I see how we use the offset, but don't understand what the offset means. Could you explain?

File: src/components/MusicalTyping.vue [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        this.$emit('stopNote', frequency);
+      },
+      keyIsAValidNote(key) {
+        return this.keys.map(k => k.letter).includes(key);
+      },
+      keyNumber(key) {
+        return key.offset + this.octave * 12;
+      },
+      frequency(keyNumber) {
+        const power = (keyNumber - 49) / 12;
```
Comment (FrankyFrankFrank) [inline src/components/MusicalTyping.vue]: yeah there's a few magic numbers here I would want to know more about

**Comment by nickdown** [comment]
@FrankyFrankFrank I extracted the magic numbers into constants and renamed `offset` to `stepsAboveRootOfOctave`. Lmk what you think!

**Comment by FrankyFrankFrank** [comment]
That's what I think

**Pull request merged**